torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -882,6 +882,12 @@ void XLAGraphExecutor::ExtractIRAndPrepareXlaData_(
       handle = WrapXlaData(xla::ComputationClient::Get()->WrapDataShards(
           {UnwrapXlaData(handle)}, GetVirtualDevice().toString(),
           sharding->shape.value(), sharding->sharding));
+    } else if (xla::sys_util::GetEnvBool("XLA_USE_SPMD", false)) {
+      // Create a placeholder PjRtReplicatedData when using SPMD
```

Comment:
hmm, so the assumption is that if we are under `SPMD` mode, all of the tensor that does not have the sharding annotation must be replicated. Is that the right reading?

Reply:
Can you remind me how virtual device is impacted.

Reply:
That's the correct reading: under SPMD mode, every tensor that's not explicitly sharded should will be replicated. cc: @yeounoh in case I'm overlooking something.

Virtual device will defer data transfer until CreateTensorsData, which in the `mark_sharding` path will be called with a ShardingSpec to distribute the tensor across local devices. If we don't call `mark_sharding`, then the data should instead be replicated. Currently, this happens in InputHandler through `CopyToDevice`, but this change moves the replication step into CreateTensorsData.